mailbox/api/src/main/java/org/apache/james/mailbox/events/Group.java

```diff
@@ -21,7 +21,27 @@
 
 import java.util.Objects;
 
+import com.google.common.base.Preconditions;
+
 public class Group {
+    public static Group deserialize(String serializedGroup) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
+        if (serializedGroup.startsWith(GenericGroup.class.getName() + GenericGroup.DELIMITER)) {
```

Comment:
Can you explain to me: 
- What `GenericGroup` implies to?
- Why the `asString` format of `GenericGroup` is different with other?

Reply:
Sure...

Basically, all groups have a different class. `GroupMailboxListener` typically exposes a `getDefaultGroup` method that allows retrieving the group to be used for this listener. Here it's easy: each listener exposes a group super class with no field. ClassName is used for equality.

However, we end up registrating several time the same listener class, and expect each listener to get a distinct group. This is for instance the case for `Quota Mailing`. We may want to set up a first listener to `trigger a 80% information` and another `trigger a 98% warning`... The above approach don't work: we need 2 different groups despite it being the same class. So we decided to configure it (James 1 & James 2 needs to use the same group for 80%, for 98%, etc...).  The `GenericGroup` holds the configured string allowing to get distinct execution groups.

So, if we get a Generic group, the format is `FQDN-configured_string` while in the generic case it is just `FQDN`